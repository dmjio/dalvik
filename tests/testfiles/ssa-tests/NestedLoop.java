class NestedLoop {
  static int entry(int x, int y) {
    int z = 1;
    for(int i = 0; i < x; ++i) {
      for(int j = 0; j < y; ++j) {
        z += j * i;
        i += 1;
      }
    }

    return z;
  }
}
