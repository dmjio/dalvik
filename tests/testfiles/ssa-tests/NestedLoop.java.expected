SimplePhi { numBlocks = 7
          , phis = [ (0, fromList ["1", "%23"])
                   , (1, fromList ["0", "%32"])
                   , (2, fromList ["%17", "%27"])
                   , (3, fromList ["%16", "%28"])
                   , (4, fromList ["0", "%29"])
                   ]
          , blockPhis = [(0, 1), (1, 1), (2, 3), (3, 3), (4, 3)]
          , assignedPhis = []
          }